store/store.go

```diff
@@ -506,6 +504,16 @@ func (s *Store) defaultSnapQuery() url.Values {
 	return q
 }
 
+func (s *Store) queryForSnapInfo() url.Values {
```

Comment:
I think would change the PR description a bit, the PR makes it sound like its about getStructFields refactor but its about more functional stuff it seems :)

Reply:
I think I failed to mention this PR was stacked on #4748...